src/auth/negotiate/negotiate_external_acl

```diff
@@ -0,0 +1,63 @@
+#!/bin/sh
+#
+# This external acl helper checks the Kerberos group information without
+# continues ldap queries by leveraging the kerbors ticket group information
+# i.e. PAC field
+#
+# It requires only during startup aldap query to get the SID for the group name.
+#
+# Squid config lines:
+# auth_param negotiate program /opt/squid/sbin/negotiate_wrapper_auth -d --ntlm /usr/bin/ntlm_auth --helper-protocol=squid-2.5-ntlmssp --domain SAMBA.HOME --kerberos /opt/squid/sbin/negotiate_kerberos_auth -d -s GSS_C_NO_NAME -k /etc/squid/squid.keytab -t none
+# external_acl_type test_acl ipv4 %LOGIN %note{group} /opt/squid/sbin/negotiate_external_acl
+# acl squid_allow external test_acl
+#
+# GET SID for Group
+#
+# Please change sample values with real valuse
+#
+if [ -z "$KRB5_KTNAME" ]; then
+   export KTNAME="/etc/krb5.keytab"
+else
+   export KTNAME=$KRB5_KTNAME
+fi
+export PRINCIPAL="HTTP/opensus42.suse.home"
+export DC="dc1.samba.home"
+export KRB5CCNAME="/tmp/squid_krb5cc"
+export GROUPNAME="SQUID_ALLOW"
+export BASEDN="DC=samba,DC=home"
+#
+#
+#
+
+kinit -kt $KTNAME $PRINCIPAL
+SID=`ldapsearch -LLL -Ygssapi -H ldap://$DC:389 -s sub -b "BASEDN" "(CN=$GROUPNAME)" objectsid 2>&1 | awk '{ if ( $0 ~/^object/ ) print $2}'`
+
+(>&2 echo "`date +"%Y/%m/%d %H:%M:%S"`| negotiate_external_acl: SID=$SID")
+
+#
+# Loop over input
+#
+while read input 
+do
+  found=0
+  user=`echo $input | awk '{ print $1 }'`
+  groups=`echo $input | awk '{ print $2 }'`
+  (>&2 echo "`date +"%Y/%m/%d %H:%M:%S"`| negotiate_external_acl: user=$user")
+  (>&2 echo "`date +"%Y/%m/%d %H:%M:%S"`| negotiate_external_acl: groups=$groups")
+  if [ -n "$groups" ]; then
+    while read group; do
```

Comment:
this little shell trick is not very obvious what is happening and requires spawning several sub-shells to execute. This would be better for both readability and performance:
  for group in $groups; do ...

Reply:
I don't think that will work if the group name has a space which is possible in AD ( I assume a  comma  is not used)

Reply:
Forget it IFS is still valid

Reply:
huh? you lost me on that last comment

Reply:
Initially, @huaraz thought that his read-based trick handles group names with spaces and your for-based iteration code does not. Then he decided that his read-based trick suffers from the same problem because the default `IFS` (input field separator) would apply to the reading command. However, IFS is _not_ applied to single-parameter `read`. At least not in `bash`.

```
rousskov@pail:~$ groups="a b,c d"

rousskov@pail:~$ while read group; do echo "group=$group"; done <<< "$(echo $groups | tr , "\n" )"
group=a b
group=c d

rousskov@pail:~$ for group in $groups; do echo "group=$group"; done
group=a
group=b,c
group=d
```